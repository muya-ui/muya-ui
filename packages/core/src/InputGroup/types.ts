import { IComponentSizeSpec } from '@muya-ui/theme-light';

import { HTMLAttributes } from 'react';

export interface IInputGroupPureProps {
  /**
   * 对InputGroup下的所有组件设置统一的尺寸
   *
   * @type {IComponentSizeSpec}
   * @memberof IInputGroupProps
   * @default 'm'
   */
  size?: IComponentSizeSpec;
  /**
   * 禁用InputGroup
   *
   * @type {boolean}
   * @default false
   * @memberof IInputGroupProps
   */
  disabled?: boolean;
}

export type IInputGroupProps = IInputGroupPureProps & HTMLAttributes<HTMLDivElement>;
