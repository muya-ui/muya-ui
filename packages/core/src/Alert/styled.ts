import styled, { css } from 'styled-components';

import { IComponentSizeSpec } from '@muya-ui/theme-light';

import { StyledButton, StyledLink } from '../Button/styled';
import { IThemedBaseProps } from '../types';
import { IAlertProps } from './types';

interface IBaseSizeProps {
  $size: IComponentSizeSpec;
}

export interface IStyledAlertProps extends IThemedBaseProps, IBaseSizeProps {
  $type: Required<IAlertProps>['type'];
  $hasTitle?: boolean;
  $hasDesc?: boolean;
  $showIcon?: boolean;
  $showClose?: boolean;
}

export interface IStyledIconProps extends IThemedBaseProps, IBaseSizeProps {
  $type: Required<IAlertProps>['type'];
  $hasTitleAndDesc?: boolean;
}

export interface IStyledCloseProps extends IThemedBaseProps, IBaseSizeProps {
  $hasTitleAndDesc?: boolean;
}

export interface IStyledTitleProps extends IThemedBaseProps, IBaseSizeProps {
  $hasDesc?: boolean;
}

export const StyledAlert = styled.div`
  ${(props: IStyledAlertProps) => {
    const { $type, $hasTitle, $hasDesc, $showIcon, $showClose, $size, theme } = props;
    const {
      typeBg,
      typeBorder,
      borderRadius,
      padding,
      hasIconPaddingLeft,
      hasIconAndDescPaddingLeft,
      hasCloseIconPaddingRight,
      hasTitleAndDescPaddingVertical,
    } = theme.components.Alert;
    let [top, right, bottom, left] = padding[$size];
    if ($showIcon) {
      left = $hasDesc ? hasIconAndDescPaddingLeft[$size] : hasIconPaddingLeft[$size];
    }
    if ($showClose) {
      right = hasCloseIconPaddingRight[$size];
    }
    if ($hasTitle && $hasDesc) {
      top = hasTitleAndDescPaddingVertical[$size];
      bottom = hasTitleAndDescPaddingVertical[$size];
    }
    return css`
      position: relative;
      box-sizing: border-box;
      background-color: ${typeBg[$type!]};
      border: ${typeBorder[$type!]};
      border-radius: ${borderRadius};

      /** 直接改 Button 存在风险，因此覆盖 */
      & ${StyledButton}, & ${StyledLink} {
        vertical-align: unset;
      }

      ${StyledAlertContainer} {
        padding: ${top}px ${right}px ${bottom}px ${left}px;
      }
    `;
  }}
`;

export const StyledAlertContainer = styled.div`
  position: relative;
  margin: 0 auto;
`;

export const StyledIcon = styled.div`
  ${(props: IStyledIconProps) => {
    const { $type, theme, $hasTitleAndDesc, $size } = props;
    const {
      left: iconLeft,
      hasTitleAndDescLeft,
      top: iconTop,
      hasTitleAndDescTop,
    } = theme.components.Alert.icon;
    const { lineHeight } = theme.components.Alert.title;
    const finalLineHeight = $hasTitleAndDesc ? 1 : `${lineHeight[$size]}px`;
    const left = $hasTitleAndDesc ? hasTitleAndDescLeft : iconLeft[$size];
    const top = $hasTitleAndDesc ? hasTitleAndDescTop : iconTop[$size];
    return css`
      position: absolute;
      left: ${left}px;
      top: ${top}px;
      color: ${theme.colors.pattern.feature[$type!]};
      font-size: 0;
      line-height: ${finalLineHeight};
    `;
  }}
`;

export const StyledClose = styled.div`
  ${(props: IStyledCloseProps) => {
    const { theme, $hasTitleAndDesc, $size } = props;
    const { closeIconTop, closeIconRight, hasTitleAndDescCloseIconTop } = theme.components.Alert;
    let finalTop = closeIconTop[$size];
    if ($hasTitleAndDesc) {
      finalTop = hasTitleAndDescCloseIconTop;
    }
    return css`
      position: absolute;
      right: ${closeIconRight}px;
      top: ${finalTop}px;
      line-height: 0;
    `;
  }}
`;

export const StyledTitle = styled.div`
  ${(props: IStyledTitleProps) => {
    const { theme, $size, $hasDesc } = props;
    const {
      colors,
      typography: {
        spec: { fontWeight },
      },
    } = theme;
    const {
      fontSize,
      lineHeight,
      textFine,
      hasDescTextFine,
      hasDescFontSize,
      hasDescLineHeight,
      hasDescMarginBottom,
    } = theme.components.Alert.title;
    const finalTextFine = $hasDesc ? hasDescTextFine : textFine;
    const finalFontSize = $hasDesc ? hasDescFontSize : fontSize[$size];
    const finalLineHeight = $hasDesc ? hasDescLineHeight : lineHeight[$size];
    const finalMarginBottom = $hasDesc ? hasDescMarginBottom : 0;
    return css`
      color: ${colors.pattern.text.title};
      font-size: ${finalFontSize}px;
      line-height: ${finalLineHeight}px;
      font-weight: ${finalTextFine ? fontWeight.regular : fontWeight.semibold};
      margin-bottom: ${finalMarginBottom}px;
    `;
  }}
`;

export const StyledDesc = styled.span`
  ${(props: IThemedBaseProps) => {
    const { theme } = props;
    const { colors } = theme;
    const { fontSize, lineHeight } = theme.components.Alert.desc;
    return css`
      color: ${colors.pattern.text.text};
      font-size: ${fontSize}px;
      line-height: ${lineHeight}px;
    `;
  }}
`;
