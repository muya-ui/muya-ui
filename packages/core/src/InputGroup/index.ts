export { default } from './InputGroup';
export { default as InputGroup } from './InputGroup';
export * from './types';
