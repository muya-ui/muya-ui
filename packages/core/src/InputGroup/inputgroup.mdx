---
name: InputGroup 输入框组
menu: 组件
route: /components/inputgroup
autoDemos: true
parent: 数据录入
---

# InputGroup 输入框组

输入框组合展现。

{Demos}

### API

#### IInputGroupProps

<Props propsName="IInputGroupProps" />
