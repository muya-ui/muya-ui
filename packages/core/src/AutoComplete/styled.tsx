import styled, { css } from 'styled-components';

import ellipsisStyle from '../styled/mixins/ellipsis';

export const StyledAutoCompleteInputWrapper = styled.span`
  position: relative;
  display: inline-block;
  width: 100%;
`;

export const StyledPirmaryField = styled.span`
  ${ellipsisStyle};
`;

export const StyledSecondaryField = styled.span`
  flex-shrink: 0;
  margin-left: 12px;
`;

export const StyledAutoCompleteItem = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  ${props => css`
    ${StyledSecondaryField} {
      color: ${props.theme.colors.pattern.text.assistant};
    }
  `};
`;
