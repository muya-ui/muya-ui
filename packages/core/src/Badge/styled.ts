import styled, { css } from 'styled-components';

import { IComponentSizeSpec } from '@muya-ui/theme-light';

import { IThemedBaseProps } from '../types';

export interface IStyledBadgeProps extends IThemedBaseProps {
  $detached: boolean;
  $isStroke: boolean;
  $value?: number | string;
  $showZero: boolean;
  $dot: boolean;
  $size: IComponentSizeSpec;
  $color?: string;
}

const badgePositionCSS = (props: IStyledBadgeProps) => {
  const { theme, $detached, $value, $showZero, $dot } = props;
  const {
    Badge: { offset, zIndex },
  } = theme.components;
  const visible = $value || $showZero || $dot;
  const scaleCSS = `scale(${visible ? 1 : 0})`;
  let positionCSS;
  if ($detached) {
    positionCSS = `
      top: 50%;
      left: 100%;
      transform: translate(${offset}px, -50%) ${scaleCSS};
    `;
  } else {
    positionCSS = `
      top: 0;
      right: 0;
      transform: translate(50%, -50%) ${scaleCSS};
    `;
  }

  return css`
    position: absolute;
    ${positionCSS}
    z-index: ${zIndex};
  `;
};

const badgeCoreCSS = (props: IStyledBadgeProps) => {
  const { theme, $size, $dot, $isStroke, $color } = props;
  const {
    components: { Badge: token },
    transition: {
      pattern: { duration, easing },
    },
  } = theme;

  let height;
  let padding;
  if ($dot) {
    height = token.height.dot[$size!];
    padding = 0;
  } else {
    height = token.height.text[$size!];
    padding = token.padding[$size!];
  }

  const fontSize = token.fontSize[$size!];
  const { borderColor } = token;
  const lineHeight = $isStroke ? height - 2 : height;
  const color = $isStroke ? token.borderColor : token.color;
  const backgroundColor = $color || token.backgroundColor[$isStroke ? 'stroke' : 'fill'];

  return css`
    display: inline-block;
    height: ${height}px;
    line-height: ${lineHeight}px;
    text-align: center;
    padding: 0 ${padding}px;
    color: ${color};
    font-size: ${fontSize}px;
    background-color: ${backgroundColor};
    border: ${$isStroke ? `1px solid ${borderColor}` : 'none'};
    border-radius: ${height / 2}px;
    white-space: nowrap;
    transform-origin: center center;
    transition: all ${duration.status}ms ${easing.status};
    ${$dot && `width: ${height}px`};
  `;
};

export const StyledBadge = styled.span`
  ${badgePositionCSS}
  ${badgeCoreCSS}
`;

export const StyledIndependentBadge = styled.span`
  ${badgeCoreCSS}
`;

// used for mock font-size less than 12px in chrome
export const StyledBadgeInnerText = styled.span`
  display: inline-block;
  font-style: normal;
  vertical-align: top;
  transform-origin: center center;
`;

export const StyledBadgeContainer = styled.span`
  display: inline-block;
  position: relative;
`;
