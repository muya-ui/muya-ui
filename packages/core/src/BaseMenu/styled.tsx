import styled, { css } from 'styled-components';

import { IComponentSizeSpec } from '@muya-ui/theme-light';

import ellipsisStyle from '../styled/mixins/ellipsis';
import { itemDisabledStyle, itemNormalStyle, itemSelectedStyle } from '../styled/mixins/item';
import { scrollBarStyle } from '../ScrollView/styled';
import { IThemedBaseProps } from '../types';

export interface IStyledBaseMenuProps extends IThemedBaseProps {
  $size?: IComponentSizeSpec;
  $width?: number;
  $maxItemCountPerPage?: number;
  $hasScrollBar: boolean;
}

export interface IStyledMenuItemProps extends IThemedBaseProps {
  $size?: IComponentSizeSpec;
  $active: boolean;
  $disabled: boolean;
  $selected: boolean;
}

export const menuItemCss = (props: IStyledMenuItemProps) => {
  const {
    $size,
    $active,
    $selected,
    $disabled,
    theme: {
      transition: {
        pattern: { duration, easing },
      },
    },
  } = props;
  const token = props.theme.components.BaseMenu.item;
  return css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex: 1;
    box-sizing: border-box;
    padding: 0 ${token.paddingHorizontal}px;
    font-size: ${token.fontSize[$size!]}px;
    line-height: ${token.height[$size!]}px;
    height: ${token.height[$size!]}px;
    cursor: pointer;
    transition: background-color ${duration.status}ms ${easing.status},
      color ${duration.status}ms ${easing.status};
    ${itemNormalStyle(token)}
    span {
      ${ellipsisStyle}
    }

    ${$active &&
      css`
        color: ${token.color.hover};
        background-color: ${token.background.hover};
        &:hover {
          color: ${token.color.hover};
          background-color: ${token.background.hover};
        }
        &:active {
          color: ${token.color.hover};
          background-color: ${token.background.hover};
        }
      `}
    ${$selected && itemSelectedStyle(token)}
    ${$disabled && itemDisabledStyle(token)}
  `;
};

export const StyledBaseMenuItem = styled.div`
  ${menuItemCss};
`;

export const StyledBaseMenuItemGroupLabel = styled.span`
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-sizing: border-box;
  cursor: default;
`;

export const StyledBaseMenuItemGroup = styled.div``;

export const StyledBaseMenuItemDivider = styled.div`
  height: 1px;
`;

export const StyledBaseMenuWrapper = styled.div`
  overflow-y: auto;
`;
export const baseMenuCss = (props: IStyledBaseMenuProps) => {
  const {
    $size,
    $width,
    $hasScrollBar,
    $maxItemCountPerPage,
    theme: { colors, shadows },
  } = props;
  const token = props.theme.components.BaseMenu;
  const {
    wrapper: wrapperToken,
    item: itemToken,
    divider: dividerToken,
    group: groupToken,
  } = token;
  return css`
    box-shadow: ${shadows.pattern.popper.normal};
    background: ${colors.pattern.background.higher};
    border-radius: ${wrapperToken.borderRadius};
    padding: ${wrapperToken.paddingVertical}px 0;
    overflow: hidden;
    ${$width && `width: ${$width}px`};
    ${$hasScrollBar && `padding-right: ${wrapperToken.paddingRight}px`};

    ${StyledBaseMenuWrapper} {
      ${$maxItemCountPerPage && `max-height: ${$maxItemCountPerPage * itemToken.height[$size!]}px`};
      ${$hasScrollBar &&
        css`
          padding-right: ${wrapperToken.innerPaddingRight}px;
          ${scrollBarStyle(props)};
        `}
    }

    ${StyledBaseMenuItemDivider} {
      background: ${colors.pattern.background.divider};
      margin: ${dividerToken.paddingVertical}px ${$hasScrollBar ? 0 : dividerToken.paddingRight}px
        ${dividerToken.paddingVertical}px ${dividerToken.paddingLeft[$size!]}px;
    }

    ${StyledBaseMenuItemGroup} {
      ${StyledBaseMenuItem} {
        padding-left: ${groupToken.paddingLeft[$size!] + groupToken.itemIndent}px;
      }
    }

    ${StyledBaseMenuItemGroupLabel} {
      font-size: ${groupToken.fontSize}px;
      line-height: ${groupToken.height}px;
      height: ${groupToken.height}px;
      background: ${groupToken.background};
      color: ${groupToken.color};
      padding-left: ${groupToken.paddingLeft[$size!]}px;
    }
  `;
};

export const StyledBaseMenu = styled.div`
  ${baseMenuCss};
`;
