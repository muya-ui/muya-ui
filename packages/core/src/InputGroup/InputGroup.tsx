import styled, { css } from 'styled-components';
import React, { useState, useCallback, CSSProperties, useMemo } from 'react';
import { IInputGroupProps } from './types';
import { IComponentSizeSpec } from '@muya-ui/theme-light';
import { StyledInputWrapper, StyledTagsInputWrapper } from '../Input/styled';
import useTheme from '../utils/useTheme';

interface ZIndexHocProps {
  children: React.ReactNode;
  size?: IComponentSizeSpec;
}

function ZIndexHoc(props: ZIndexHocProps) {
  const { children, size } = props;
  const childProps = (children as any).props;
  const childType = (children as any).type;
  const [focus, setFocus] = useState(false);
  const [popupVisible, setPopupVisible] = useState(false);
  const theme = useTheme();
  const handleFocus = useCallback(
    (e: React.MouseEvent) => {
      if (childProps.onFocus) {
        childProps.onFocus(e);
      }
      setFocus(true);
    },
    [childProps],
  );

  const handleBlur = useCallback(
    (e: React.MouseEvent) => {
      if (childProps.onBlur) {
        childProps.onBlur(e);
      }
      setFocus(false);
    },
    [childProps],
  );

  const handlePopupVisibleChange = useCallback(
    (open: boolean) => {
      if (childProps.onPopupVisibleChange) {
        childProps.onPopupVisibleChange(open);
      }
      setPopupVisible(open);
    },
    [childProps],
  );

  // 组件处于focus或者popupVisible状态，提升zIndex层级
  const shouldAddZIndex = focus || popupVisible;
  const childStyle: CSSProperties = useMemo(
    () => ({
      zIndex: shouldAddZIndex ? 1 : undefined,
    }),
    [shouldAddZIndex],
  );

  const finalChildProps: any = useMemo(
    () => ({
      onFocus: handleFocus,
      onBlur: handleBlur,
      size: childProps.size || size,
    }),
    [childProps.size, handleBlur, handleFocus, size],
  );

  // 只针对Select相关组件添加此事件，否则会出现warning
  // Select相关组件blur后实际上还处于聚焦状态，需要特殊处理保证zIndex正确
  if (
    childType.__MUYA_TREESELECT ||
    childType.__MUYA_SELECT ||
    childType.__MUYA_CASCADER ||
    childType.__MUYA_AUTOCOMPLETE
  ) {
    finalChildProps.onPopupVisibleChange = handlePopupVisibleChange;
  }

  // 设置plain为true，即为线性按钮，带有border
  // type为normal的情况下，默认也是为线性按钮
  const isPlainButton =
    !theme.components.Button.plainIsLight &&
    (childProps.plain || !childProps.type || childProps.type === 'normal');
  // 非线性按钮，不需要向右移动
  if (childType.__MUYA_BUTTON && !isPlainButton) {
    childStyle.marginRight = 0;
  }

  finalChildProps.style = {
    ...childStyle,
    ...childProps.style,
  };

  return React.cloneElement(children as any, finalChildProps);
}

const StyledInputGroup = styled.div<IInputGroupProps>`
  ${props => {
    const compactStyle = props.disabled
      ? undefined
      : css`
          & > * {
            position: relative;
            float: none;
            vertical-align: top;

            &:focus {
              z-index: 1;
            }

            /* 中间节点，统一去除border-radius */
            &:not(:first-child):not(:last-child) {
              &,
              & ${StyledInputWrapper}, & ${StyledTagsInputWrapper} {
                border-radius: 0;
              }
            }

            /* 首节点右侧border-radius处理 */
            &:first-child {
              &,
              & ${StyledInputWrapper}, & ${StyledTagsInputWrapper} {
                border-top-right-radius: 0;
                border-bottom-right-radius: 0;
              }
            }

            /* 尾节点左侧border-radius处理 */
            &:last-child {
              &,
              & ${StyledInputWrapper}, & ${StyledTagsInputWrapper} {
                border-top-left-radius: 0;
                border-bottom-left-radius: 0;
              }
            }

            /* 除了尾节点的所有节点，向右移动1px，解决双重border的问题 */
            &:not(:last-child) {
              margin-right: -1px;
              border-right-width: 1px;
            }
          }
        `;

    return css`
      position: relative;
      display: block;
      width: 100%;
      ${compactStyle}
    `;
  }}
`;

const InputGroup = React.memo((props: IInputGroupProps) => {
  const { children, size, disabled = false, ...other } = props;
  return (
    <StyledInputGroup disabled={disabled} {...other}>
      {React.Children.map(children, child => {
        if (!React.isValidElement(child)) {
          return null;
        }

        return <ZIndexHoc size={size}>{child}</ZIndexHoc>;
      })}
    </StyledInputGroup>
  );
});

export default InputGroup;
