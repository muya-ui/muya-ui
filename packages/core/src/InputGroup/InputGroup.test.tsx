import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import Button from '../Button';
import Input from '../Input';
import { StyledPopperInputWrapper } from '../Input/styled';
import Select, { Option } from '../Select';
import InputGroup from './InputGroup';

test('should render correctly', () => {
  const tree = renderer
    .create(
      <InputGroup>
        <Input></Input>
        <Button>确认</Button>
      </InputGroup>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('should set zIndex when focus and blur', () => {
  const handleFocus = sinon.spy();
  const handleBlur = sinon.spy();
  const handleVisibleChange = sinon.spy();
  const wrapper = mount(
    <InputGroup>
      <Input autoFocus id="test1" />
      <Select id="select1" onPopupVisibleChange={handleVisibleChange}>
        <Option value="111" />
      </Select>
      <Button>确认</Button>
    </InputGroup>,
  );

  const wrapperWithCallback = mount(
    <InputGroup disabled>
      <Input onBlur={handleBlur} onFocus={handleFocus} autoFocus id="test2" />
      <Button type="primary">确认</Button>
      {''}
    </InputGroup>,
  );

  wrapper
    .find('input')
    .at(0)
    .simulate('focus', {});

  expect(
    window.getComputedStyle(
      wrapper
        .find('#test1')
        .at(0)
        .getDOMNode(),
    ).zIndex,
  ).toBe('1');

  wrapper
    .find('input')
    .at(0)
    .simulate('blur', {});

  expect(
    window.getComputedStyle(
      wrapper
        .find('#test1')
        .at(0)
        .getDOMNode(),
    ).zIndex,
  ).toBe('');

  wrapperWithCallback
    .find('input')
    .at(0)
    .simulate('focus', {});
  wrapperWithCallback
    .find('input')
    .at(0)
    .simulate('blur', {});
  wrapper
    .find(StyledPopperInputWrapper)
    .find('input')
    .at(0)
    .simulate('click', {});

  expect(handleFocus.calledOnce).toBe(true);
  expect(handleBlur.calledOnce).toBe(true);
  expect(handleVisibleChange.calledOnce).toBe(true);
});
