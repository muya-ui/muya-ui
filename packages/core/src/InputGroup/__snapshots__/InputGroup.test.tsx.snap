// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.c4 {
  outline: 0;
  margin: 0;
  font-family: inherit;
  text-transform: none;
  overflow: visible;
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 32px;
  padding: 0 13px 0 13px;
  font-weight: 600;
  font-size: 14px;
  line-height: 20px;
  border-radius: 2px;
  color: rgba(102,102,102,1);
  background-color: transparent;
  cursor: pointer;
  border-color: rgba(204,204,204,1);
  border-style: solid;
  border-width: 1px;
}

.c6:hover {
  color: rgba(102,102,102,1);
  background-color: rgba(102,102,102,0.04);
  border-color: rgba(204,204,204,1);
}

.c6:active {
  color: rgba(102,102,102,1);
  background-color: rgba(102,102,102,0.08);
  border-color: rgba(204,204,204,1);
}

.c6 + .c5 {
  margin-left: 8px;
}

.c2 {
  width: 240px;
  height: 32px;
  padding: 0 12px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  line-height: 16px;
  border-radius: 2px;
  border: 1px solid rgba(221,221,221,1);
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  vertical-align: middle;
  background-color: rgba(255,255,255,1);
}

.c2:hover {
  border-color: rgba(209,209,209,1);
}

.c3 {
  display: inline-block;
  outline: none;
  border: 0;
  padding: 0;
  margin: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
  background: none;
  font-size: 12px;
  line-height: 16px;
  caret-color: rgba(26,122,248,1);
  color: rgba(51,51,51,1);
}

.c3::selection {
  background: rgba(26,122,248,0.16);
}

.c3::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c3::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c3:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c3::placeholder {
  color: rgba(204,204,204,1);
}

.c0 {
  position: relative;
  display: block;
  width: 100%;
}

.c0 > * {
  position: relative;
  float: none;
  vertical-align: top;
}

.c0 > *:focus {
  z-index: 1;
}

.c0 > *:not(:first-child):not(:last-child),
.c0 > *:not(:first-child):not(:last-child) .c1,
.c0 > *:not(:first-child):not(:last-child) .StyledTagsInputWrapper-kkbZlb {
  border-radius: 0;
}

.c0 > *:first-child,
.c0 > *:first-child .c1,
.c0 > *:first-child .StyledTagsInputWrapper-kkbZlb {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.c0 > *:last-child,
.c0 > *:last-child .c1,
.c0 > *:last-child .StyledTagsInputWrapper-kkbZlb {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.c0 > *:not(:last-child) {
  margin-right: -1px;
  border-right-width: 1px;
}

<div
  className="c0"
  disabled={false}
>
  <div
    className="c1 c2 muya-input-input-wrapper"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    size="m"
    style={
      Object {
        "zIndex": undefined,
      }
    }
  >
    <input
      className="c3 muya-input-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="m"
      value=""
    />
  </div>
  <button
    className="c4 c5 c6"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "zIndex": undefined,
      }
    }
    type="button"
  >
    чбошод
  </button>
</div>
`;
